packages/server/src/unstable-core-do-not-import/procedureBuilder.ts

```
@@ -624,6 +629,24 @@
+
+      await _def.asyncStorage.run(opts.ctx, async () => {
+        // Process the middleware chain without asyncStorage to prevent هnfinite recursion
+        result = await callRecursive(
+          0,
+          { ..._def, asyncStorage: undefined },
+          opts,
+        );
+      });
+
+      // Return the result from the async context
+      return result!;
```

ESLint

[error] Forbidden non-null assertion.

(@typescript-eslint/no-non-null-assertion)